src/app/core/store/documents/documents.filters.ts

```diff
@@ -198,13 +209,22 @@ function filterDocumentsByStem(
 }
 
 function cleanStemForBaseCollection(stem: QueryStem, documents: DocumentModel[]): QueryStem {
-  return cleanStemForCollection(stem, documents, stem.collectionId);
+  return cleanStemForCollectionAndLink(stem, documents, stem.collectionId);
```

Comment:
I have seen the same method on the backend and I just wonder, what do these clean methods do?

Reply:
The idea is to split `QueryStem` into separate stems which contains only concrete collection (and its link) based `filters`, `linkFilters` and `documentIds`. Then I'm filtering documents in multiples stages from the first collection (called as "base collection") to the latest collection in `QueryStem`.